meson.build

```diff
@@ -0,0 +1,302 @@
+  'src/opus_multistream_encoder.c',
+  'src/opus_multistream_decoder.c',
+  'src/repacketizer.c',
+  'src/opus_projection_encoder.c',
+  'src/opus_projection_decoder.c',
+  'src/mapping_matrix.c'
+]
+
+opus_sources_float = [
+  'src/analysis.c',
+  'src/mlp.c',
+  'src/mlp_data.c'
+]
+
+opus_sources += opus_sources_float
+
+celt_sources = [
+  'celt/bands.c',
+  'celt/celt.c',
+  'celt/celt_encoder.c',
+  'celt/celt_decoder.c',
+  'celt/cwrs.c',
+  'celt/entcode.c',
+  'celt/entdec.c',
+  'celt/entenc.c',
+  'celt/kiss_fft.c',
+  'celt/laplace.c',
+  'celt/mathops.c',
+  'celt/mdct.c',
+  'celt/modes.c',
+  'celt/pitch.c',
+  'celt/celt_lpc.c',
+  'celt/quant_bands.c',
+  'celt/rate.c',
+  'celt/vq.c'
+]
+
+celt_sources_sse = [
+  'celt/x86/x86cpu.c',
+  'celt/x86/x86_celt_map.c',
+  'celt/x86/pitch_sse.c'
+]
+
+celt_sources_sse2 = [
+  'celt/x86/pitch_sse2.c',
+  'celt/x86/vq_sse2.c'
+]
+
+celt_sources_sse4_1 = [
+  'celt/x86/celt_lpc_sse4_1.c',
+  'celt/x86/pitch_sse4_1.c'
+]
+
+cc = meson.get_compiler('c')
+
+if cc.get_argument_syntax() == 'msvc'
+  sse4_1_arg = ''
+else
+  sse4_1_arg = '-msse4.1'
+endif
+
+sse = get_option('sse')
+if sse.auto()
+  have_sse = cc.links('''
+    #include <xmmintrin.h>
+    #include <time.h>
+    int main() {
+      __m128 mtest;
+      mtest = _mm_set1_ps((float)time(NULL));
+      mtest = _mm_mul_ps(mtest, mtest);
+      return _mm_cvtss_si32(mtest);
+    }''')
+else
+  have_sse = sse.enabled()
+endif
+
+sse2 = get_option('sse2')
+if sse2.auto()
+  have_sse2 = cc.links('''
+    #include <emmintrin.h>
+    #include <time.h>
+    int main() {
+      __m128i mtest;
+      mtest = _mm_set1_epi32((int)time(NULL));
+      mtest = _mm_mul_epu32(mtest, mtest);
+      return _mm_cvtsi128_si32(mtest);
+    }''')
+else
+  have_sse2 = sse2.enabled()
+endif
+
+sse4_1 = get_option('sse4_1')
+if sse4_1.auto()
+  have_sse4_1 = cc.links('''
+    #include <smmintrin.h>
+    #include <time.h>
+    int main() {
+      __m128i mtest;
+      mtest = _mm_set1_epi32((int)time(NULL));
+      mtest = _mm_mul_epi32(mtest, mtest);
+      return _mm_cvtsi128_si32(mtest);
+    }''',
+    args: sse4_1_arg)
+else
+  have_sse4_1 = sse4_1.enabled()
+endif
+
+cdata = configuration_data()
+cdata.set('OPUS_BUILD', 1)
+
+if have_sse
+  message('SSE intrinsics available')
+  celt_sources += celt_sources_sse
+endif
+
+if have_sse2
+  message('SSE 2 intrinsics available')
+  celt_sources += celt_sources_sse2
+endif
+
+if have_sse4_1
+  message('SSE 4.1 intrinsics available')
+  silk_sources += silk_sources_sse4_1
+  celt_sources += celt_sources_sse4_1
+  cdata.set('OPUS_X86_MAY_HAVE_SSE4_1', 1)
+  add_project_arguments(sse4_1_arg, language: 'c')
+endif
+
+if cc.compiles('''
+    int main() {
+      static int x; char a[++x]; a[sizeof a - 1] = 0; int N; return a[0];
+    }''', name: 'C99 variable-size arrays')
+  cdata.set('VAR_ARRAYS', 1)
+elif build_machine.system() == 'windows'
+  message('Making use of alloca()')
+  cdata.set('USE_ALLOCA', 1)
```

Comment:
Shouldn't use else to using alloca() anywhere that VAR_ARRAYS don't work and alloca exists? Maybe you want to do something like this?
```
elif cc.has_function('alloca')
```

Reply:
`cc.has_function('alloca')` doesn't seem to work with MSVC, so an additional explicit compiler check is needed:

```
elif cc.get_argument_syntax() == 'msvc' or cc.has_function('alloca')
```